clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -798,7 +798,7 @@ public void recordTopicFetchMetrics(String topic, int bytes, int records) {
             Sensor recordsFetched = this.metrics.getSensor(name);
             if (recordsFetched == null) {
                 Map<String, String> metricTags = new HashMap<>(1);
-                metricTags.put("topic", topic.replace('.', '_'));
+                metricTags.put("topic", topic);
```

Comment:
I synced up with @junrao offline about this. Originally it was introduced in the core package's metrics since Graphite treats "." as hierarchy separators; for the client metrics, it is supposed that the GraphiteReported should take care of this itself rather than letting Kafka metrics to special handle for it.

In addition, right now only consumer Fetcher had replace, but producer Sender does not have it actually.